src/goto-instrument/full_slicer.cpp

```diff
@@ -402,6 +402,8 @@ void full_slicert::operator()(
       const exprt &s=to_code_dead(e_it->first->code).symbol();
       decl_dead[to_symbol_expr(s).get_identifier()].push(e_it->second);
     }
+    else if(e_it->first->is_function_call())
+      add_to_queue(queue, e_it->second, e_it->first);
```

Comment:
This implies that all function calls are preserved, even when not on any path to the actual slicing criterion. I do agree that a variant of this is required for soundness (a major improvement would require changes as found, e.g., in Jens Krinke's PhD thesis), but ideally that would be limited to only those function calls on the call graph where the criterion can possibly be reached.